synapse/config/emailconfig.py

```diff
@@ -108,9 +107,14 @@ def read_config(self, config, **kwargs):
             if self.trusted_third_party_id_servers:
                 # XXX: It's a little confusing that account_threepid_delegate_email is modified
                 # both in RegistrationConfig and here. We should factor this bit out
-                self.account_threepid_delegate_email = self.trusted_third_party_id_servers[
-                    0
-                ]  # type: Optional[str]
```

Comment:
I removed `Optional` from this as `self.account_threepid_delegate_email` won't be assigned here if `self.trusted_third_party_id_servers` is falsey. But would mypy take this hint and apply it even if the code won't run?

Reply:
I think you meant `account_threepid_delegate_email`, but TIL, thanks!